validator/client/validator_propose.go

```diff
@@ -61,6 +73,12 @@ func (v *validator) ProposeBlock(ctx context.Context, slot uint64, pubKey [48]by
 		Signature: sig,
 	}
 
+	history = SetProposedForEpoch(history, epoch)
```

Comment:
I think this should be after `v.validatorClient.ProposeBlock` in L83.  Only if beacon node has successfully broadcast the block on validator's behave then we set it to the `ProposalHistory`

Reply:
I set it before since it's safer. If for any reason the validator were to error upon saving, it would have already proposed the block and wouldn't know.

Reply:
It's not safer. It's the same. Not sure what you mean by validator error upon saving. Block only gets broadcasted upon a successful response from beacon node. This logic breaks the failover part and disallow proposer to propose to the back up beacon node